runtime/vara/src/lib.rs

```diff
@@ -1108,6 +1114,62 @@ impl pallet_vesting::Config for Runtime {
     const MAX_VESTING_SCHEDULES: u32 = 28;
 }
 
+#[derive(Encode, Decode, Clone, Eq, PartialEq, Default, TypeInfo)]
+pub struct VoucherLegitimate;
+
+impl SignedExtension for VoucherLegitimate {
+    const IDENTIFIER: &'static str = "VoucherLegitimate";
```

Comment:
Could you explain please the reason of this additional validation? In order to save money for somebody who uses voucher wrong?